### PR TITLE
Bulk upload validation FTE pay check - warning if two or more workers in the same job role have the same annual salary but different hours

### PR DESCRIPTION
**Pull request by g-m-taylor**
#### Tests
Does this PR include tests for the changes introduced?
- [x] Yes
- [ ] No, I found it difficult to test
- [ ] No, they are not required for this change

### CODE REVIEW

File: server/routes/establishments/bulkUpload.js [hunk 1]
```diff
@@ -2473,8 +2476,53 @@ const checkDuplicateLocations = async (
       locations[locationId] = thisEstablishment.lineNumber;
     });
 };
+// check if hours matches others in the same job and same annual pay
+const checkPartTimeSalary = (thisWorker, myWorkers, csvWorkerSchemaErrors) => {
+  if (thisWorker._currentLine.STATUS === 'UNCHECKED' || thisWorker._currentLine.STATUS === 'DELETE') {
+    return;
+  }
+  if (
+    (thisWorker._currentLine.CONTHOURS !== '' && thisWorker._currentLine.CONTHOURS < 37) &&
```
Comment (aaron-russell) [inline server/routes/establishments/bulkUpload.js:2485]: Would CONTHOURS be a number?

File: server/routes/establishments/bulkUpload.js [hunk 2]
```diff
@@ -2473,8 +2476,53 @@ const checkDuplicateLocations = async (
...
 
-const checkDuplicateWorkerID = (thisWorker, allKeys, changeKeyNoWhitespace, keyNoWhitespace, allWorkersByKey, myAPIWorkers, csvWorkerSchemaErrors ) => {
+    if (otherWorkerFTE) {
+      csvWorkerSchemaErrors.push(thisWorker.ftePayCheckHasDifferentHours());
+    } else if (workersToCheckinDB.length) {
+      if (myCurrentEstablishments.some(function(establishment) {
+        return workersToCheckinDB.some(function(localID) {
+          localID = localID.replace(/\s/g, '');
+          if (establishment._workerEntities[localID]) {
+            const worker = establishment._workerEntities[localID.replace(/\s/g, '')];
```
Comment (aaron-russell) [inline server/routes/establishments/bulkUpload.js:2510]: You have already applied the Regex?

File: server/routes/establishments/bulkUpload.js [hunk 3]
```diff
@@ -2473,8 +2476,53 @@ const checkDuplicateLocations = async (
...
+            if ((worker.annualHourlyPay && worker.annualHourlyPay.value === 'Annually' && worker.annualHourlyPay.rate == thisWorker._currentLine.SALARY) && (worker.mainJob)) {
```
Comment (aaron-russell) [inline server/routes/establishments/bulkUpload.js:2511]: You shouldn't need the brackets round `worker.mainJob`

File: server/test/unit/routes/establishments/bulkUpload.spec.js [hunk 4]
```diff
@@ -181,6 +181,381 @@ describe('/server/routes/establishment/bulkUpload.js', () => {
...
+          origin: 'Workers',
+          lineNumber: 1,
+          errCode: 998,
+          errType: 'DUPLICATE_ERROR',
+          error: 'CHGUNIQUEWRKID Worker 1 is not unique',
+          name: 'foo',
+          source: 'Worker 2',
+          worker: 'Worker 2'
+        });
+      });
```
Comment (aaron-russell) [inline server/test/unit/routes/establishments/bulkUpload.spec.js:558]: Where did this come from? 😂

File: server/routes/establishments/bulkUpload.js (hunk 1 above)
Comment (g-m-taylor) [reply server/routes/establishments/bulkUpload.js]: I added parseFloat to explicitly change it to a float. does that make more sense?

**Pull request merged**